tools/c7n_azure/c7n_azure/query.py

```diff
@@ -92,9 +93,13 @@ def filter(self, resource_manager, **params):
         results = []
         for parent in parents.resources():
             if extra_args:
-                params.update({key: parent[extra_args[key]] for key in extra_args.keys()})
+                params.update({key: extra_args[key](parent) for key in extra_args.keys()})
+
+            if enum_op:
+                op = getattr(getattr(client, enum_op), list_op)
+            else:
```

Comment:
enum_op and list are synonyms... please comment the distinction, and if they are truly different one should not be a default for the other.